model/version.go

```diff
@@ -356,6 +371,42 @@ func VersionGetHistory(versionId string, N int) ([]Version, error) {
 	return versions, nil
 }
 
+type MainlineCommitVersionOptions struct {
```

Comment:
`MainlineCommitsOptions` is a generated struct based off of the graphql schema input. I dont think this makes sense to consolidate even though the fields are similar there are some differences. Also if we try to extend the schema input it would affect the query options and add fields that both the resolver or the aggregation might not need.